include/util.h

```diff
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <math.h>
```

Comment:
You should be able to use M_PI (not standard, but basically every compiler implements it...) instead of PI down below.

I'd still prefer PI to be constexpr instead of a `#define`, if you manually define it.